### DISCUSSION TITLE
How to cache to file?

### DISCUSSION THREAD

**Discussion by vuminhquang** [ThreeMammals/Ocelot#2001] [Q&A]
Hello,
If I want to use CacheManager to cache to file, with no expiration, instead of memory, do anyone know where to start to code this?

Thank you

**Comment by catcherwong** [comment]
At first, you should ensure that CacheManager can write cache data to file, if not, you should do this by yourself.

What's more, overwrite `Ocelot.Cache.CacheManager` this project, to change the `ExpirationMode`

https://github.com/ThreeMammals/Ocelot/blob/41519b1e5649f1ae85be81353f9f8b7c0aef85b7/src/Ocelot.Cache.CacheManager/OcelotCacheManagerCache.cs#L15-L18

**Comment by vuminhquang** [comment]
@catcherwong : thank you for answering me. Seems that  CacheManager don't have feature which writes cache data to file, I will try to write a new CacheManager service for Ocelot.
Since I am not familiar with IoC, I am very appreciate if you could show me (when you have spare time) how to inject the new service in Startup.cs. Thank you.

**Reply by raman-m** [reply]
The community will help you.
Just wait for a couple of years :wink:

**Comment by catcherwong** [comment]
You can take a look on the following repository, it may help you how to do.

https://github.com/catcherwong/Ocelot.Cache.EasyCaching

**Reply by raman-m** [reply]
Perfect!
But how does it cache to file? ))

**Comment by vuminhquang** [comment]
Thank you very much @catcherwong , brilliant code.
I'll study it.
